pkg/certagent/agent.go

```diff
@@ -108,17 +112,30 @@ func GenerateCSRObject(config CSRConfig) (*capi.CertificateSigningRequest, error
 
 // RequestCertificate will create a certificate signing request for a node
 // with the config given and send it to a signer via a POST request.
-// If something goes wrong it returns an error.
+// If something goes wrong it returns an error unless CSRConfig.MaxRetry
+// is defined in which case we will retry the request.
 // NOTE: This method does not return the approved CSR from the signer.
 func (c *CertAgent) RequestCertificate() error {
+	waitRetry := 10 * time.Second
+	try.MaxRetries = c.config.MaxRetry
+
 	csr, err := GenerateCSRObject(c.config)
 	if err != nil {
 		return fmt.Errorf("error generating CSR Object: %v", err)
 	}
 
-	// send CSR to the signer
-	if _, err := c.client.Create(csr); err != nil {
-		return fmt.Errorf("error sending CSR to signer: %v", err)
+	// implement a retry mechanism in the case request fails.
+	errc := try.Do(func(attempt int) (bool, error) {
```

Comment:
why not https://godoc.org/k8s.io/apimachinery/pkg/util/wait#ExponentialBackoff compared to pulling in new dependency?

Reply:
ok makes sense will adjust.